dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -6576,10 +6595,10 @@ int processWhere(SELECT_LEX &select_lex,
         else if (select_lex.where)
             icp = select_lex.where;
     }
-    else if (!join && ( ((gwi.thd->lex)->sql_command == SQLCOM_UPDATE ) ||
```

Comment:
I would put all boilerplate conditions into inline bool functions to reduce the code.

Reply:
Good point. I have added inline bool functions to reduce this code repetition.